### PR TITLE
Fix infomax

### LINKED ISSUE #6919: BUG?  Extended infomax ICA
This looks like a bug to me, the documentation says that (through a backwards incompatible change) `if you want Extended Infomax, set method='infomax' and fit_params=dict(extended=True)`, however for `method='infomax'` [these lines](https://github.com/mne-tools/mne-python/blob/master/mne/preprocessing/ica.py#L359-L360) force `extended` to `False`.

### PR DESCRIPTION
**Pull request by drammock**
closes #6919

### CODE REVIEW

**Comment by larsoner** [comment]
+1 for merge and backport once CIs are happy

File: mne/preprocessing/ica.py [hunk 1]
```diff
@@ -357,10 +357,10 @@ def __init__(self, n_components=None, max_pca_components=None,
...
                                not in fit_params})
         elif method == 'infomax':
-            fit_params.update({'extended': False})
-        elif method == 'extended-infomax':
-            fit_params.update({'extended': True})
-            method = 'infomax'
+            # extended=True is default in underlying function, but we want
+            # default False here unless user specified True:
+            ext = fit_params.get('extended', False)
+            fit_params.update({'extended': ext})
```
Comment (christianbrodbeck) [inline mne/preprocessing/ica.py]: Could use a single line 
```
            fit_params.setdefault('extended', False)
```

Comment (drammock) [reply mne/preprocessing/ica.py]: ooh, that is nicer. Thanks.

**Comment by agramfort** [comment]
back ported to 0.19 branch

**Pull request merged**